doc/release/v2_3_72.md

```diff
@@ -21,6 +21,13 @@ New Features
 * Added the property *yarprun* to ports opened through `yarprun --server`.
 * Added the property *nameserver* for the port opened through `yarpserver`.
 * The method `yarp::os::Network::runNameServer()` was removed.
+* Clock refactoring: clock type can be initialized in the yarp::os::Network constructor or
+  in the Network::init() function. This will take precedence over the environment variable.
+  yarp::os::SystemRateThread class created as RateThread clone but runs using system clock
+  only, independently from other settings.
+  Any call to yarp::os::Time::now() or delay() before the object yarp::os::Netowork is
+  initialized will thrown an assert to avoid undefined behaviour of uninitialized Time object
```

Comment:
Sorry if I missed this point when checking the PR, but this change is **extremely abrupt** and will break a lot of third party code. I notice it because random tests in iDynTree started to fail, but I quickly checked and all this examples/tutorials are now broken: 
* https://github.com/robotology/icub-tutorials/blob/4d5d7bba87fb2d7a90d35c0b06746e4696e1918a/src/iKin/fwInvKinematics/main.cpp
* https://github.com/robotology/icub-tutorials/blob/4d5d7bba87fb2d7a90d35c0b06746e4696e1918a/src/optimization/calibReference/main.cpp
* https://github.com/robotology/icub-tutorials/blob/4d5d7bba87fb2d7a90d35c0b06746e4696e1918a/src/learningMachines/lmlibportable.cpp
* https://github.com/robotology/icub-tutorials/blob/4d5d7bba87fb2d7a90d35c0b06746e4696e1918a/doc/hello_icub.dox
* https://github.com/robotology/yarp/blob/e0ca8e6f73c37286734d06e3afb8015529ee36b9/example/profiling/timers.cpp

Failing early is definitely more robust, but I think this is the perfect example of the case in which we should respect our deprecation policy ( https://github.com/robotology/yarp/blob/master/.github/CONTRIBUTING.md#deprecated-features ), i.e. first printing a deprecation warning when using `yarp::os::Time::now()` and only after at least one release actually remove the functionality. Otherwise, we should definitely remove that section from the Contributing guide, at least to be honest with our users! 

cc @drdanz @barbalberto